includes/cart/functions.php

```diff
@@ -122,73 +122,118 @@ function edd_add_to_cart( $download_id, $options = array() ) {
+		if ( isset( $options[ 'quantity' ] ) && is_array( $options[ 'quantity' ] ) ) {
+			$quantity = $options[ 'quantity' ];
+		} else if ( isset( $options[ 'quantity' ] ) ) {
+				$quantity = array( $options[ 'quantity' ] );
+			} else {
+			$quantity = array( 1 );
+			$options[ 'quantity' ] = $quantity;
+		}
+
+		if ( is_array( $options[ 'quantity' ] ) ) {
+			// if the number of quantities is the same as the number of price options, match them up
+			if ( count( $options[ 'quantity' ] ) == count( $options[ 'price_id' ] ) ) {
+				foreach ( edd_combine_array( $options[ 'price_id' ], $options[ 'quantity' ] ) as $price => $quantity ) {
+					$local_options = $options;
+					$local_options[ 'price_id' ] = preg_replace( '/[^0-9\.-]/', '', $price );
+					$local_options[ 'quantity' ] = $quantity;
+
+					$last_cart_count = edd_add_to_cart( $download_id, $local_options );
+				}
+			} else if ( count( $options[ 'quantity' ] ) > count( $options[ 'price_id' ] )  ) { // if there's more quantities than price_ids, match up the first pairs of each, ignoring extra quantities
+					foreach ( edd_combine_array( $options[ 'price_id' ], $options[ 'quantity' ] ) as $price => $quantity ) {
+						$local_options = $options;
+						$local_options[ 'price_id' ] = preg_replace( '/[^0-9\.-]/', '', $price );
+						$local_options[ 'quantity' ] = $quantity;
+
+						$last_cart_count = edd_add_to_cart( $download_id, $local_options );
+					}
+				} else { // else match them up until we run out, then assume all remaining ones use the last quantity
+				$count = 0;
+				$max = count( $options[ 'quantity' ] ) - 1;
+				foreach ( $options[ 'price_id' ] as $price ) {
+					if ( $count > $max ) {
+						$quantity = $options[ 'quantity' ][ $max ];
+					} else {
+						$quantity = $options[ 'quantity' ][ $count ];
+					}
+					$count++;
+					$local_options = $options;
+					$local_options[ 'price_id' ] = preg_replace( '/[^0-9\.-]/', '', $price );
+					$local_options[ 'quantity' ] = $quantity;
+
+					$last_cart_count = edd_add_to_cart( $download_id, $local_options );
+				}
+			}
+		} else { // there is 1 quantity for all price_ids
+			foreach ( $options[ 'price_id' ] as $price ) {
+				$local_options = $options;
+				$local_options[ 'price_id' ] = preg_replace( '/[^0-9\.-]/', '', $price );
+				$local_options[ 'quantity' ] = $quantity;
+
+				$last_cart_count = edd_add_to_cart( $download_id, $local_options );
+			}
+		}
+		// done adding each price_id. Return
+		return $last_cart_count;
+	}
 
 	do_action( 'edd_pre_add_to_cart', $download_id, $options );
 
 	$cart = apply_filters( 'edd_pre_add_to_cart_contents', edd_get_cart_contents() );
 
-	if ( edd_has_variable_prices( $download_id )  && ! isset( $options['price_id'] ) ) {
+	if ( edd_has_variable_prices( $download_id )  && ! isset( $options[ 'price_id' ] ) ) {
 		// Forces to the first price ID if none is specified and download has variable prices
-		$options['price_id'] = '0';
+		$options[ 'price_id' ] = '0';
```

Comment:
`0` is no longer the first price ID. May want to use the default price ID for a download?

`edd_get_default_variable_price( $download_id )`